transports/azure-service-bus/configuration.md

```diff
@@ -34,5 +34,5 @@ These settings control how the transport connects to the broker.
 
  * `UseWebSockets()`: Configures the transport to use AMQP over websockets.
  * `TimeToWaitBeforeTriggeringCircuitBreaker(TimeSpan)`: The time to wait before triggering the circuit breaker after a critical error occurred. Defaults to 2 minutes.
- * `CustomTokenProvider(ITokenProvider)`: Allows replacement of the default token provider, which uses the shared secret in the connection string for authentication. This opens up additional authentication mechanisms such as [shared access signatures](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-sas), [SAML, Oauth, SWT, windows authentication](https://docs.microsoft.com/en-us/dotnet/api/microsoft.servicebus.tokenprovider?view=azure-dotnet), [managed identities for Azure resources](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-managed-service-identity), or even custom implementations.
-partial: custom-retry-policy
\ No newline at end of file
+ * `CustomTokenCredential(TokenCredential)`: Allows usage of Azure Active Directory (AAD) authentication instead of the shared secret in the connection string. This opens up additional authentication mechanisms such as [managed identities for Azure resources](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-managed-service-identity). When leveraging AAD authentication, it is worth noting that the underlying library expects a fully-qualified namespace (like `<asb-namespace-name>.servicebus.windows.net`) instead of a connection string (like `Endpoint=sb://<asb-namespace-name>.servicebus.windows.net>;[...]`).
```

Comment:
```suggestion
 * `CustomTokenCredential(TokenCredential)`: Enables using a non-default authentication mechanism such as Azure Active Directory (AAD) or [managed identities for Azure resources](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-managed-service-identity) instead of the shared secret stored in the connection string. 

Note: AAD authentication requires a fully-qualified namespace usage (e.g. `<asb-namespace-name>.servicebus.windows.net`) instead of a connection string (e.g. `Endpoint=sb://<asb-namespace-name>.servicebus.windows.net>;[...]`).
```

Reply:
Looks good overall, thanks.

> Enables using a non-default authentication mechanism such as Azure Active Directory (AAD) or managed identities for Azure resources

A nitpick here is that using managed identities is using AAD authentication, so the "or" reads a bit strange to me.
In other words, we're using AAD authentication when we're using managed identities; however, they're not the only way to use AAD authentication, we could use a "regular" service principal (client id / client secret), or the developer's identity when running a solution locally, for example.

I would reword your suggestion as "Enables using Azure Active Directory (AAD) authentication such as managed identities for Azure resources instead of the shared secret in the connection string."

How do you think that reads?

Reply:
@mderriey-prospa sounds good. I've added your tweaks to the suggestion. Please let me know if that's good and I'll tweak the changes to be version specific.

Reply:
Looks hood @tmasternak, thanks for that 😉